pkg/job/discovery.go

```diff
@@ -210,10 +212,19 @@ func getFilteredMetricDatas(
 	dimensionNameList []string,
 	m *config.Metric,
 ) []*model.CloudwatchData {
-	associator := newMetricsToResourceAssociator(dimensionRegexps, resources)
+	type resourceAssociator interface {
+		AssociateMetricsToResources(cwMetric *cloudwatch.Metric) (*model.TaggedResource, bool)
```

Comment:
Nit: AssociateMetricsToResources -> AssociateMetricToResource just to make it more obvious it's 1 to 1 at most